libraries/chain/steem_evaluator.cpp

```diff
@@ -563,19 +565,25 @@ namespace golos { namespace chain {
                     } else if (_db.is_producing()) {
                         max_depth = STEEMIT_SOFT_MAX_COMMENT_DEPTH;
                     }
-                    FC_ASSERT(parent->depth < max_depth,
-                              "Comment is nested ${x} posts deep, maximum depth is ${y}.",
-                              ("x", parent->depth)("y", max_depth));
+                    GOLOS_CHECK_LOGIC(parent->depth < max_depth,
+                            logic_exception::reached_comment_max_depth,
+                            "Comment is nested ${x} posts deep, maximum depth is ${y}.",
+                            ("x", parent->depth)("y", max_depth));
                 }
                 auto now = _db.head_block_time();
 
                 if (itr == by_permlink_idx.end()) {
                     if (o.parent_author != STEEMIT_ROOT_POST_PARENT) {
-                        FC_ASSERT(_db.get(parent->root_comment).allow_replies, "The parent comment has disabled replies.");
+                        GOLOS_CHECK_LOGIC(_db.get(parent->root_comment).allow_replies, 
+                                logic_exception::replies_are_not_allowed,
+                                "The parent comment has disabled replies.");
+
                         if (_db.has_hardfork(STEEMIT_HARDFORK_0_12__177) && !_db.has_hardfork( STEEMIT_HARDFORK_0_18__536) ) {
-                            FC_ASSERT(
+                            GOLOS_CHECK_LOGIC(
```

Comment:
No chance to get here: all current operations have no restriction and all pre-HF18 ops are already valid. @afalaleev, maybe we should remove this `if` completely?

Reply:
Yes, you're right, this check can be removed.